src/fios.cpp

```diff
@@ -205,12 +205,16 @@ const char *FiosBrowseTo(const FiosItem *item)
  */
 static void FiosMakeFilename(char *buf, const char *path, const char *name, const char *ext, const char *last)
 {
-	const char *period;
+	assert(*path != '\0' && *name != '\0');
+
+	buf = strecpy(buf, path, last);
+	/* Remove trailing path separator, if present */
+	if (buf[-1] == PATHSEPCHAR) buf--;
```

Comment:
I'm not a fan of this. Negative array indices are scary, and I think in the case that buf == last, this could lead to an out-of-bounds access